ledger/src/blockstore_db.rs

```diff
@@ -1439,11 +1446,17 @@ impl<'a> WriteBatch<'a> {
         self.map[C::NAME]
     }
 
-    pub fn delete_range_cf<C: Column>(
+    /// Adds a \[`from`, `to`) range deletion entry to the batch.
+    ///
+    /// Note that the \[`from`, `to`) deletion range of WriteBatch::delete_range_cf
+    /// is different from \[`from`, `to`\] of Database::delete_range_cf as we makes
+    /// the semantics of Database::delete_range_cf matches the blockstore purge
+    /// logic.
+    fn delete_range_cf<C: Column>(
```

Comment:
@yhchiang-sol  I'm very happy about our new consistent interval handling with `[from, to]`. :)

that said, I think we should put the range manipulation code as deep as possible for encapsulation.

so, i think this non-`pub` fn might be a good place to actually `.saturating_add(1)`. That's because it looks like `WriteBatch::delete_range_cf` is only called by `Database::delete_range_cf`?

Then, we can remove these rather extra justification comment about different semantics put at the docstring in `WriteBatch::delete_range_cf`

is there strong reason we're adjusting the `to` at `Database::delete_range_cf` specifically?

Reply:
The type here is C::Index, which could be u64, (u64, u64), Pubkey, Signature, (u64, Signature, Slot), and (u64, Pubkey, Slot, Signature).  We will need to implement the C::saturating_add for each of them.  Might be worth-trying I think, although some types might be tricky.

If we want to move lower to the rocksdb delete_range_cf, then is more difficult to perform +1 as it takes arbitrary byte array.

A cleaner solution is to make RocksDB's range delete optionally perform the inclusive deletion based on the WriteOptions where we will add a new boolean indicating inclusive deletion, but I guess this would take a much longer route as we need to carry this information into range-deletion key format and update the internal range-deletion logic to honor this.

So probably good for now I think.